planner/core/logical_plan_builder.go

```diff
@@ -136,6 +140,12 @@ func (b *PlanBuilder) buildAggregation(p LogicalPlan, aggFuncList []*ast.Aggrega
 	plan4Agg.GroupByItems = gbyItems
 	plan4Agg.SetSchema(schema4Agg)
 	plan4Agg.collectGroupByColumns()
+	if hint := b.TableHints(); hint != nil {
+		plan4Agg.preferAggType = hint.preferAggType
+	}
+	if bits.OnesCount(plan4Agg.preferAggType) > 1 {
```

Comment:
Can we check this earlier?

Reply:
For example? 0.0

Reply:
For example? 0.0

Reply:
For example?

Reply:
Can we put this in line 91?

Reply:
It seems we don't need this check any more. Just throw a warning in `exhaustPhysicalPlan` is OK.

Reply:
Why don't we throw the warning here? This line is touched earlier than exhaustPhysicalPlan.

Reply:
I think it's doesn't matter? Do we need to throw warnings as early as possible? I don't know that, I put it in `exhaustPhysicalPlan` because I think it's easier to read.